honeybee_energy/ventcool/afn.py

```diff
@@ -0,0 +1,391 @@
+
+            * ext_walls: List of exterior Wall type Face objects.
+            * ext_roofs: List of exterior RoofCeiling type Face objects.
+            * ext_apertures: List of exterior Aperture subface Face objects.
+            * ext_doors: List of exterior Door subface Face objects.
+
+        ext_cracks: A dictionary of air mass flow coefficient and exponent data
+            corresponding to the face types in the exterior_face_groups. Face
+            data flow coefficients should be normalized by surface area, and closed
+            opening flow coefficients should be normalized by edge lengths, for example:
+
+            {
+                # kg/s/m2 @ 1 Pa
+                'wall_flow_cof': 0.00001,
+                'wall_flow_exp': 0.7,
+                'roof_flow_cof': 0.00001,
+                'roof_flow_exp': 0.70,
+                # kg/s/m @ 1 Pa
+                'window_flow_cof': 0.00001,
+                'window_flow_exp': 0.7,
+                'door_flow_cof': 0.0002,
+                'door_flow_exp': 0.7
+            }
+    """
+
+    # simplify parameters
+    ext_walls, ext_roofs, ext_apertures, ext_doors = exterior_face_groups
+
+    # add exterior crack leakage components
+    for ext_wall in ext_walls:
+        opening_area = sum([aper.area for aper in ext_wall.apertures])
+        opening_area += sum([door.area for door in ext_wall.doors])
+        face_area = ext_wall.area - opening_area
+        flow_cof = ext_cracks['wall_flow_cof'] * face_area
+        flow_exp = ext_cracks['wall_flow_exp']
+        ext_wall.properties.energy.vent_crack = AFNCrack(flow_cof, flow_exp)
+
+    for ext_roof in ext_roofs:
+        opening_area = sum([aper.area for aper in ext_roof.apertures])
+        face_area = ext_roof.area - opening_area
+        flow_cof = ext_cracks['roof_flow_cof'] * face_area
+        flow_exp = ext_cracks['roof_flow_exp']
+        ext_roof.properties.energy.vent_crack = AFNCrack(flow_cof, flow_exp)
+
+    # add exterior opening leakage components
+    for ext_aperture in ext_apertures:
+        if ext_aperture.properties.energy.vent_opening is None:
+            ext_aperture.is_operable = True
+            ext_aperture.properties.energy.vent_opening = VentilationOpening()
+        vent_opening = ext_aperture.properties.energy.vent_opening
+        vent_opening.flow_coefficient_closed = ext_cracks['window_flow_cof']
+        vent_opening.flow_exponent_closed = ext_cracks['window_flow_exp']
+
+    for ext_door in ext_doors:
+        if ext_door.properties.energy.vent_opening is None:
+            ext_door.properties.energy.vent_opening = VentilationOpening()
+        vent_opening = ext_door.properties.energy.vent_opening
+        vent_opening.flow_coefficient_closed = ext_cracks['door_flow_cof']
+        vent_opening.flow_exponent_closed = ext_cracks['door_flow_exp']
+
+
+def generate(rooms, window_vent_controls, leakage_type='Average',
+             use_room_infiltration=True, atmospheric_pressure=101325):
```

Comment:
I'm thinking this should be the only public method in this whole `afn.py` module.